### PR TITLE
GS USB Streaming

### PR DESCRIPTION
**Pull request by jbinz**
### Description
This patch implements a USB streaming interface to the ground station.

The data Interface looks as follows:

Link Num [1,2] Timestamp in s, Flight state, Latitude, Longitude, Altitude, Velocity, Battery Voltage

and is updated with 10 Hz if data is received with 10 Hz. 

### Checklist for changes

- [x] Feature tested manually [Mandatory]
- [ ] Simulation flight performed [Optional]
- [ ] Drone flight performed [Optional]
- [ ] Copyright notice for new files
- [ ] `#pragma once` directive for new header files

### Checklist for Releases
- [ ] Simulation flight performed [Mandatory]
- [ ] Drone flight performed [Optional]
- [ ] Version number incremented [Mandatory]
- [ ] Develop and debug flag disabled [Mandatory]

### Related issues
#347

### CODE REVIEW

File: ground_station/src/telemetry/parser.cpp [hunk 1]
```diff
@@ -83,5 +82,5 @@ void Parser::cmdGNSSTime(uint8_t *args, uint32_t length) {
 
 // NOLINTNEXTLINE(readability-convert-member-functions-to-static) function ptr doesn't work if it's static
 void Parser::cmdGNSSInfo(uint8_t *args [[maybe_unused]], uint32_t length [[maybe_unused]]) {
-  console.log.println("GNSS Info Received");
+  // To Do: Do Something
```
Comment (stojadin2701) [inline ground_station/src/telemetry/parser.cpp:85]: TODO

File: ground_station/src/utils.cpp [hunk 2]
```diff
@@ -202,6 +186,28 @@ int32_t Utils::getFlashMemoryUsage() {
...
+  const auto lat_prec = static_cast<int32_t>((link->data.lat() - static_cast<float>(lat_int)) * 100000.0F);
+
+  const auto lon_int = static_cast<int32_t>(link->data.lon());
+  const auto lon_prec = static_cast<int32_t>((link->data.lon() - static_cast<float>(lon_int)) * 100000.0F);
+
+  const auto voltage_int = static_cast<int32_t>(link->data.voltage());
+  const auto voltage_prec = static_cast<int32_t>((link->data.voltage() - static_cast<float>(voltage_int)) * 10.0F);
+
+  std::snprintf(print_char, 150,
+                "Link %d: Ts: %ld.%ld, State: %d, Lat: %ld.%05ld, Lon: %ld.%05ld, Alt: %ld, Vel: %d, V: %ld.%ld",
```
Comment (stojadin2701) [inline ground_station/src/utils.cpp:205]: can you paste some sample output

File: ground_station/src/main.cpp [hunk 3]
```diff
@@ -43,6 +41,9 @@ void setup() {
 
 void loop() {
   static bool ini{false};
+  static uint16_t link1LastTs{0};
```
Comment (stojadin2701) [inline ground_station/src/main.cpp:44]: we have 16-bit timestamps?

File: ground_station/src/utils.cpp (hunk 2 above)
Comment (jbinz) [reply ground_station/src/utils.cpp:205]: image

File: ground_station/src/main.cpp (hunk 3 above)
Comment (jbinz) [reply ground_station/src/main.cpp:44]: Yeah, the telemetry sends down 16 bit timestamps

File: ground_station/src/utils.hpp [hunk 4]
```diff
@@ -8,6 +8,8 @@
 
 #pragma once
 
+#include "telemetry/telemetry.hpp"
```
Comment (stojadin2701) [inline ground_station/src/utils.hpp:11]: just forward-declare the telemetry struct/class

below all includes:

 struct Telemetry; (if struct)
 class Telemetry; (if class)

File: ground_station/src/main.cpp [hunk 5]
```diff
@@ -25,9 +25,14 @@ void setup() {
   pinMode(21, INPUT);
 
   console.begin();
+#ifdef CATS_DEBUG
+  console.setLevel(Console::ConsoleLevel::LEVEL_OK);
```
Comment (stojadin2701) [inline ground_station/src/main.cpp:29]: "OK" level is warning level? 😅

File: ground_station/src/main.cpp [hunk 6]
```diff
@@ -67,29 +75,51 @@ void loop() {
...
-      if (link2DataValid) {
+      if (link2GpsValid) {
         navigation.setPointB(link2.data.lat(), link2.data.lon());
-      } else if (link1DataValid) {
+      } else if (link1GpsValid) {
         navigation.setPointB(link1.data.lat(), link1.data.lon());
       }
     }
   } else {
+    // Stream both Links as they track different FC's
```
Comment (stojadin2701) [inline ground_station/src/main.cpp:107]: FC's -> FCs

**Review by stojadin2701 — APPROVED** [review]

**Pull request merged**